web/app/components/document/sidebar.ts

```diff
@@ -477,6 +507,23 @@ export default class DocumentSidebarComponent extends Component<DocumentSidebarC
     return !this.editingIsDisabled;
   }
 
+  /**
+   * The action to show the Projects modal.
+   * Triggered by clicking the "+" button in the Projects section.
+   */
+  @action protected showProjectsModal() {
+    this.projectsModalIsShown = true;
+  }
+
+  /**
+   * The action to hide the Projects modal.
+   * Passed to the Projects modal component and
+   * triggered on modal close.
+   */
+  @action protected hideProjectsModal() {
+    this.projectsModalIsShown = false;
+  }
+
```

Comment:
Non-blocking: maybe could simplify to a `toggleProjectsModall()` that does `this.projectsModalIsShown = !this.projectsModalIsShown`?

Reply:
I usually reserve the `toggle` construction for actual toggles, e.g., "Show more/Show fewer" or check/uncheck controls. Here, because the show and hide actions are separate, I like the clarity of being specific: <button {{on "click" this.showProjectsModal}}>` and `<Hds::Modal @onClose={{this.hideProjectsModal}}>` feels more precise to me than if we were to pass `toggleProjectsModal` to both.

Also, it's not the case here, but show/hide actions often have specific setup or teardown, so I might also be in the habit of separating them from the jump.